packages/gov-app/pages/identity/connect.tsx

```diff
@@ -0,0 +1,272 @@
+import type { NextPage } from "next";
+import { Layout } from "@gov-app/libs/components/Layout";
+import { Header } from "@gov-app/libs/components/Header";
+import {
+	ChangeEventHandler,
+	FormEventHandler,
+	MouseEventHandler,
+	useCallback,
+	useState,
+} from "react";
+import { useWindowPopup } from "@gov-app/libs/hooks/useWindowPopup";
+import { Button } from "@gov-app/libs/components/Button";
+import { ReactComponent as DiscordLogo } from "@gov-app/libs/assets/vectors/discord.svg";
+import { ReactComponent as TwitterLogo } from "@gov-app/libs/assets/vectors/twitter.svg";
+import { ReactComponent as CENNZLogo } from "@gov-app/libs/assets/vectors/cennz.svg";
+import { ReactComponent as Spinner } from "@gov-app/libs/assets/vectors/spinner.svg";
+import { TextField } from "@gov-app/libs/components/TextField";
+import { Select } from "@gov-app/libs/components/Select";
+import { useCENNZExtension } from "@gov-app/libs/providers/CENNZExtensionProvider";
+import { useCENNZWallet } from "@gov-app/libs/providers/CENNZWalletProvider";
+import { getSession } from "next-auth/react";
+import { ReactComponent as CloseIcon } from "@gov-app/libs/assets/vectors/close-icon.svg";
+import { If } from "react-extras";
+
+const Connect: NextPage = () => {
+	const {
+		onCENNZConnectClick,
+		onCENNZAccountSelect,
+		allAccounts,
+		selectedAccount,
+	} = useCENNZConnect();
+
+	const {
+		onSignInClick: onTwitterSignInClick,
+		username: twitterUsername,
+		clearUsername: clearTwitterUsername,
+	} = useSocialSignIn("Twitter");
+	const {
+		onSignInClick: onDiscordSignInClick,
+		username: discordUsername,
+		clearUsername: clearDiscordUsername,
+	} = useSocialSignIn("Discord");
+
+	const { busy, onFormSubmit } = useFormSubmit();
+
+	// eslint-disable-next-line @typescript-eslint/no-empty-function
+	const onTextInput = useCallback(() => {}, []);
+
```

Comment:
Can this be removed?

Reply:
ah no we cannot unfortunately if I didn't put it in, React will complain about controlled input doesn't have a change event. I can change it to `readOnly` field, but then the form validation would not work 🤦